packages/authentication/src/types.ts

```diff
@@ -70,6 +83,12 @@ export interface AuthenticationStrategy {
    */
   name: string;
 
+  /**
+   * The 'securitySpec' property allows a security scheme to be provided for the
+   * strategy that will be registered as an OAS spec enhancer.
+   */
+  securitySpec?: () => SecurityScheme;
```

Comment:
I am a bit confused. The `name` property is defined and implemented as a prototype property (see e.g. `JWTAuthenticationStrategy#name`), but `securitySpec` is defined as a prototype property but implemented as a static property in your examples. As a result, the definition here is disconnected from the actual security classes.

Based on the implementation below, it seems that we need to access the security spec before the strategy class is instantiated. This makes sense to me, in that case I think we need to rework this property to be a static one.

Also, I don't understand why is this a getter function (`() => SecurityScheme`) and not a value property?

```suggestion
  securitySpec?: SecurityScheme;
```

Reply:
I was a bit confused but tried to apply some feedback rightly or wrongly.  Happy to revert to value property. 👍

Reply:
I suggested a function or a static property in https://github.com/strongloop/loopback-next/pull/4693#discussion_r387795232 so @dougal83 tried it, but the new approach looks much better than my original proposal, which creates a separate class for enhancer.
Thanks @bajtos and @dougal83